fdbserver/ClusterController.actor.cpp

```diff
@@ -709,9 +709,24 @@ class ClusterControllerData {
 		for(int i = 0; i < proxies.size(); i++)
 			result.proxies.push_back(proxies[i].interf);
 
-		auto oldLogRouters = getWorkersForRoleInDatacenter( dcId, ProcessClass::LogRouter, req.maxOldLogRouters, req.configuration, id_used );
-		for(int i = 0; i < oldLogRouters.size(); i++) {
-			result.oldLogRouters.push_back(oldLogRouters[i].interf);
+		if(req.maxOldLogRouters > 0) {
+			auto oldLogRouters = tlogs;
+			
+			if(oldLogRouters.size() == 1) {
+				result.oldLogRouters.push_back(oldLogRouters[0].interf);
+			} else {
+				bool foundCC = false;
+				for(int i = 0; i < oldLogRouters.size() - 1; i++) {
+					if(oldLogRouters[i].interf.locality.processId() != clusterControllerProcessId) {
+						result.oldLogRouters.push_back(oldLogRouters[i].interf);
+					} else {
+						foundCC = true;
+					}
+				}
+				if(foundCC) {
+					result.oldLogRouters.push_back(oldLogRouters.back().interf);
+				}
```

Comment:
So this logic means that we will always have 1 fewer old log router than tlogs (except when there's only 1)? Is there any reason not to use all of the tlogs when none of them are the cluster controller?

If what's written is the intent, the else block could be simplified to something like:

```
for(int i = 0; i < oldLogRouters.size() && result.oldRouters.size() < oldLogRouters.size() - 1; ++i) {
	if(oldLogRouters[i].interf.locality.processId() != clusterControllerProcessId) {
		result.oldLogRouters.push_back(oldLogRouters[i].interf);
	}
}